lib/Install.js

```diff
@@ -1,86 +1,71 @@
 // @flow
 
-const child_process = require('child_process');
+const spawn = require('cross-spawn');
 const fs = require('fs');
 const path = require('path');
 const rimraf = require('rimraf');
-const Compiler = require('./Compile');
+const ElmJson = require('./ElmJson');
+const Project = require('./Project');
 
-function install(pathToElmBinary /*: string */, packageName /*: string */) {
-  var oldSourceDirectories;
+void Project;
 
-  const dirPath = path.join(
-    Compiler.getGeneratedCodeDir(process.cwd()),
-    'install'
-  );
+function install(
+  project /*: typeof Project.Project */,
+  pathToElmBinary /*: string */,
+  packageName /*: string */
+) /*: 'SuccessfullyInstalled' | 'AlreadyInstalled' */ {
+  const generatedCodeDir = path.join(project.generatedCodeDir, 'install');
```

Comment:
```suggestion
  const installationScratchDir = path.join(project.generatedCodeDir, 'install');
```

Just because the phrase `generatedCodeDir` already means the parent dir of this dir so using a different name (ignore my suggestion if you have a better idea) helps disambiguate them.

Reply:
Fixed!